src/Schema/Types/Scalars/DateTimeUtc.php

```diff
@@ -17,6 +17,6 @@ protected function format(Carbon $carbon): string
     protected function parse(mixed $value): Carbon
     {
         // @phpstan-ignore-next-line We know the format to be good, so this can never return `false`
-        return Carbon::createFromIsoFormat('YYYY-MM-DDTHH:mm:ss.SSSSSSZ', $value);
+        return Carbon::createFromIsoFormat('YYYY-MM-DDTHH:mm:ss.SSSSSSZ', (string) $value);
```

Comment:
I think we should be able to change the interface for `parse` to accept `string` and validate the values beforehand in `DateScalar::tryParsingDate`.